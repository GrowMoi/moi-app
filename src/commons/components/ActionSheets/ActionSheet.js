import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, Modal, TouchableOpacity, Animated } from 'react-native';
import styled from 'styled-components/native';
import { Palette } from '../../styles';

const Overlay = styled(TouchableOpacity)`
  background-color: ${Palette.black.alpha(0.2).css()};
  justify-content: flex-end;
  align-self: stretch;
  flex: 1;
`;

const OptionsBox = styled(Animated.View)`
  background-color: white;
  position: relative;
`;

const Option = styled(TouchableOpacity)`
  padding-horizontal: 20;
  padding-vertical: 13;
`;

const TextCancel = styled(Text)`
  color: red;
`;

const options = [
  'GalerÃ­a',
  'Recomendaciones',
  'Perfil',
  'Links Relacionados',
  'Notas',
];

export default class ActionSheet extends Component {
  state = {
    translate: new Animated.ValueXY({ x: 0, y: 400 }),
  }

  dismiss = () => {
    const { onDismiss } = this.props;
    if (onDismiss) {
      setTimeout(() => {
        onDismiss();
      }, 150);
    }
    this.triggerOptions('hide');
  }

  onShow = () => {
    this.triggerOptions();
  }

  triggerOptions(state) {
    const { translate } = this.state;
    const valueY = state === 'hide' ? 400 : 0;

    Animated.spring(translate, {
      toValue: { x: 0, y: valueY },
      duration: 100,
    }).start();
  }

  render() {
    const { translate } = this.state;
    const { visible } = this.props;

    const transitionStyles = {
      transform: translate.getTranslateTransform(),
    };

    return (
      <Modal
        animationType='none'
        transparent
        visible={visible}
        onRequestClose={this.dismiss}
        onShow={this.onShow}
      >
        <Overlay onPress={this.dismiss} activeOpacity={1}>
          <OptionsBox style={transitionStyles}>
            {options.map((option, i) => (
              <Option key={i}>
                <Text>{option}</Text>
              </Option>
            ))}
            <Option onPress={this.dismiss}>
              <TextCancel>Cancel</TextCancel>
            </Option>
          </OptionsBox>
        </Overlay>
      </Modal>
    );
  }
}

ActionSheet.defaultProps = {
  visible: true,
};

ActionSheet.propTypes = {
  visible: PropTypes.bool,
  onDismiss: PropTypes.func,
};
