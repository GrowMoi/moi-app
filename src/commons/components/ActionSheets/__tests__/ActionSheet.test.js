import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components/native';
import ActionSheet from '../ActionSheet';

const options = [
  { label: 'Galeria', icon: 'md-photos' },
  { label: 'Recomendaciones', icon: 'md-thumbs-up' },
  { label: 'Links', icon: 'md-link' },
  { label: 'Notas', icon: 'md-create' },
];

describe('Action Sheet', () => {
  it('it should render as expected', () => {
    const wrapper = renderer.create(
      <ActionSheet
        visible
        options={options}
        hasCancelOption />
    );
    wrapper.toJSON();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
});
