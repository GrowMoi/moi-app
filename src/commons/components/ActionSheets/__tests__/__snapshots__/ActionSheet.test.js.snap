// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Sheet it should render as expected 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  onShow={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "rgba(0,0,0,0.2)",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "white",
          "marginHorizontal": 0,
          "position": "relative",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 400,
            },
          ],
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 13,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4a4b21",
                "fontFamily": "Futura",
                "fontSize": 15,
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Galeria
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 13,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4a4b21",
                "fontFamily": "Futura",
                "fontSize": 15,
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Recomendaciones
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 13,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4a4b21",
                "fontFamily": "Futura",
                "fontSize": 15,
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Links
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 13,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4a4b21",
                "fontFamily": "Futura",
                "fontSize": 15,
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Notas
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderTopColor": "rgba(129,129,128,0.3)",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4a4b21",
                "fontFamily": "Futura",
                "fontSize": 15,
                "textAlign": "left",
              },
              undefined,
            ]
          }
        >
          Cancelar
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
